pkg/publisher/publisher.go

```diff
@@ -73,6 +76,30 @@ func FindIface(ip net.IP) (iface net.Interface, err error) {
 	return iface, fmt.Errorf("Couldn't find interface with IP address %s", ip)
 }
 
+// If an interface changes after we've started listening on it (say because it
+// got bridged), we need to be restarted to listen on the new interface that
+// contains the specified IP.
+func IfaceCheck(ip net.IP, iface net.Interface) {
+	for {
+		i, err := FindIface(ip)
+		if err != nil || i.Name != iface.Name {
+			// The interface we were listening on changed,
+			// exit so we can be restarted on the correct one.
+			log.Printf("mdns-publish: Bound interface changed, exiting.")
+			process, err := os.FindProcess(os.Getpid())
+			if err != nil {
+				// This shouldn't be possible, but we need to handle it anyway
+				log.Printf("[ERR] mdns-publish: Failed to find own process")
+			}
+			err = process.Signal(syscall.SIGTERM)
```

Comment:
Isn't it better to pass shutdown channel here and close is internally? Instead of looking for your own PID and commit suicide. Would save some lines.

Reply:
I believe closing the shutdown channel would only stop the listening goroutine. The main thread blocks waiting for a signal (<-sig in https://github.com/openshift/mdns-publisher/blob/master/cmd/publish.go#L93). If we only close shutdown then we'd stop listening but keep running until a signal is delivered.

I also considered just calling Exit() here, but apparently that's the nuclear option and doesn't allow any cleanup to run. I don't know whether that would cause trouble, but it seemed like a graceful shutdown was less likely to cause problems.

Reply:
I see, fair enough.